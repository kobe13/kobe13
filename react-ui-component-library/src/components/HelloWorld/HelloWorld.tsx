import * as React from 'react';
import styled, { BackgroundColorValues, sdkTheme, ThemeProvider } from '../../theme';

export type AttributeProps = {
  message?: string;
};

export type StyledProps = {
  /**
   * ```
   * BackgroundColorValues {
   *    primaryColor = 'black',
   *    alternativeColor = 'grey',
   * }
   * ```
   */
  background?: BackgroundColorValues;
};

const StyledHelloWorld = styled.div<StyledProps>`
  background-color: ${(props) => props.background};
  padding: 20px;
  color: ${(props) =>
    props.background === props.theme.backgroundColor.alternativeColor
      ? props.theme.color.primaryColor
      : props.theme.color.alternativeColor};
`;

type Props = AttributeProps & StyledProps;

/**
 * This component allows to display a greeting.
 */
const HelloWorld: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <ThemeProvider theme={sdkTheme}>
      <StyledHelloWorld background={props.background}>Hello {props.message}!</StyledHelloWorld>
    </ThemeProvider>
  );
};

HelloWorld.defaultProps = {
  message: 'World',
  background: sdkTheme.backgroundColor.primaryColor,
};

export default HelloWorld;
