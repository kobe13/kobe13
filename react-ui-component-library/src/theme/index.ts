import * as styledComponents from 'styled-components';

const {
  default: styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<ThemeInterface>;

export enum BackgroundColorValues {
  primaryColor = 'black',
  alternativeColor = 'grey',
}

export enum ColorValues {
  primaryColor = 'black',
  alternativeColor = 'white',
}

export enum Radius {
  m = 15,
}

export enum Spacing {
  xs = 3,
  s = 5,
  m = 10,
  l = 15,
  xl = 20,
}

export type ThemeInterface = {
  backgroundColor: {
    primaryColor: string;
    alternativeColor: string;
  };

  color: {
    primaryColor: string;
    alternativeColor: string;
  };

  radius: {
    m: number;
  };

  spacing: {
    xs: number;
    s: number;
    m: number;
    l: number;
    xl: number;
  };
};

export const sdkTheme = {
  backgroundColor: BackgroundColorValues,
  color: ColorValues,
  radius: Radius,
  spacing: Spacing,
};

export { css, createGlobalStyle, keyframes, ThemeProvider };
export default styled;
