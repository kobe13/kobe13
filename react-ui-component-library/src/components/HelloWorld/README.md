Basic example:

```jsx
<HelloWorld />
```

Example with props:

```jsx
import { sdkTheme } from '../../theme';

<HelloWorld message="John Doe" background={sdkTheme.backgroundColor.alternativeColor} />;
```
