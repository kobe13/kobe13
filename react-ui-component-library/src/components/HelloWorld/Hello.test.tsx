import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import HelloWorld from './HelloWorld';
import { sdkTheme } from '../../theme';

describe('HelloWorld', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(' content renders correctly', () => {
    const tree = create(
      <HelloWorld message="Click me" background={sdkTheme.backgroundColor.alternativeColor} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
