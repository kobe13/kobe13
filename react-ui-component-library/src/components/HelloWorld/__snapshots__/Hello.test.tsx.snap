// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld  content renders correctly 1`] = `
<div
  className="sc-bdVaJa dKFFjg"
>
  Hello 
  Click me
  !
</div>
`;
